frontend/app/routes/api/apply-state.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * An API route that can be used to perform actions with user's apply state.
+ */
+import type { ActionFunctionArgs } from '@remix-run/node';
+import { json } from '@remix-run/node';
+
+import { z } from 'zod';
+
+import { saveApplyState } from '~/route-helpers/apply-route-helpers.server';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('routes/api/apply-state');
+
+export enum ApiApplyStateAction {
+  Extend = 'extend',
+}
+
+const bodySchema = z.object({
+  action: z.nativeEnum(ApiApplyStateAction),
+  id: z.string(),
+});
+
+export async function action({ context: { session }, request }: ActionFunctionArgs) {
+  log.debug("Action with with user's apply state");
+
+  if (request.method !== 'POST') {
+    log.warn('Invalid method requested [%s]; responding with 405', request.method);
+    throw json({ message: 'Method not allowed' }, { status: 405 });
+  }
+
+  const requestBody = await request.json();
+  const parsedBody = bodySchema.safeParse(requestBody);
+
+  if (!parsedBody.success) {
+    log.debug('Invalid request body [%j]', requestBody);
+    return json({ errors: parsedBody.error.flatten().fieldErrors }, { status: 400 });
+  }
+
+  const params = { id: parsedBody.data.id };
+
+  switch (parsedBody.data.action) {
+    case ApiApplyStateAction.Extend: {
+      log.debug("Extending user's apply state; id: [%s]", params.id);
```

Comment:
It might be worth logging the `id` of the `session` (in addition to the apply state id) to help identify any events that touch a particular session. We currently do this elsewhere like in `apply-route-helpers.server`:
https://github.com/DTS-STN/canadian-dental-care-plan/blob/43c0a65a9150f19dee8bd8d0ac096fb13b0deaff/frontend/app/route-helpers/apply-route-helpers.server.ts#L201

Reply:
Good idea. Fixed